torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -460,6 +550,51 @@ c10::optional<PointwiseParams> getPointwiseHeuristics(
     }
   }
 
+  // Vectorizing innermost domains
+
+  // Don't try to vectorize if it's not recommended
+  params.inner_factor = 1;
+
+  // Vectorize as much as we can
+  size_t vectorize_factor = max_unroll_factor;
+
+  auto vectorizable_inputs_outputs_entry =
+      HeuristicSummaryEntry<HeuristicCompileTime::VectorizableInputsAndOutputs>(
+          data_cache, [&largest_out]() {
+            return std::make_unique<std::vector<TensorView*>>(
+                scheduler_utils::getInputsOutputsWithInnerDim(
+                    largest_out, true));
+          });
+
+  auto& vectorizable_inputs_outputs = vectorizable_inputs_outputs_entry.get();
+
+  for (auto tv : vectorizable_inputs_outputs) {
+    const auto tv_vectorize_factor = runtime_info.getVectorizableWidth(tv);
+    vectorize_factor = std::min(vectorize_factor, tv_vectorize_factor);
+  }
+
+  // Try expanding vectorization to contig merged domains
+  auto expanded_vector_word_size = expandVectorizationToContigMergedDomains(
+      fusion,
+      runtime_info,
+      vectorizable_inputs_outputs,
+      largest_out,
+      break_point,
+      vectorize_factor);
+  expanded_vector_word_size = std::min(
+      static_cast<size_t>(max_unroll_factor), expanded_vector_word_size);
+  if (expanded_vector_word_size > vectorize_factor) {
+    vectorize_factor = expanded_vector_word_size;
```

Comment:
Is it a safe assumption that the pointwise scheduler will always merge the innermost contig domains before any other actions? It looks like this has to be the case if `expanded_vector_word_size` is used.

Reply:
Yes, that's assumed and I believe it is the case.